### ISSUE TITLE
ENHACEMENT: identifier field for features

### ISSUE THREAD

**Issue by mainakae** [caseypt/GeoJSON.js#15]
Hello again,

In the GeoJSON specification, it's contemplated the possibility of having an identifier field in the GeoJSON objects, stored in the `id` field. It would be great to have **GeoJSON.js** accept another parameter to specify which field should be parsed into that identifier. As of now, it can be done by iterating the results and moving the id field from properties to the feature.

Thanks for your time,
Pedro.

**Comment by jkhoel** [comment]
I have modified my local copy to account for this. It sets feature no.1 as id: 0, no.2 as 1 etc..
I could do a pull request for this change - and in my use case it would suffice. However I wanted to ask if anyone has ever had a need for this ID beeing anything other than just a sequencial number.

**Comment by tormi** [comment]
> However I wanted to ask if anyone has ever had a need for this ID beeing anything other than just a sequencial number.

Yes, this could come from input `json` and should be dealt with as `include` for `properties`.

**Comment by theshanergy** [comment]
Just in case anyone else stumbles on this in need of a solution, here is how to move the 'id' field out of properties and put it in the feature root. It would be nice if GeoJSON.js handled this natively however.

```js
// Parse data into GeoJSON.
const geojson = GeoJSON.parse(data, { Point: ['lat', 'lng'] })

// Move 'id' from properties to feature.
geojson.features = geojson.features.map(({ properties: { id, ...otherProperties }, ...rest }) => ({ id, properties: otherProperties, ...rest }))
```
